src/server/game/AuctionHouse/AuctionHouseListing.cpp

```diff
@@ -0,0 +1,598 @@
+/*
+ * Copyright (C) 2008-2017 TrinityCore <http://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AuctionHouseListing.h"
+#include "AuctionHouseMgr.h"
+#include "Creature.h"
+#include "GameEventMgr.h"
+#include "Log.h"
+#include "ObjectAccessor.h"
+#include "ObjectMgr.h"
+#include "Player.h"
+#include "GameTime.h"
+
+#include <boost/thread/locks.hpp>
+#include <boost/thread/mutex.hpp>
+
+#include <vector>
+#include <unordered_map>
+
+namespace
+{
+    std::atomic<bool> _auctionHouseListingAllowed;
+
+    std::list<AuctionListingEvent*> _requestsNew;
```

Comment:
I've removed mutex because all auction packets are processed as thread unsafe so no concurrency issues, its more like single producer / single consumer as of now and they cannot work concurrently